markdown/report.Rmd

```diff
@@ -4,8 +4,82 @@ output:
     reference_docx: "report-template.docx"
 ---
 
+```{r load-data, echo=FALSE, message=FALSE, warning=FALSE,error=FALSE}
+
+# Source setup environment and function scripts
+source(here::here("code", "00_setup-environment.R"))
+source(here::here("functions", "summarise_data.R"))
+source(here::here("functions", "calculate_qom.R"))
+source(here::here("functions", "day_diff.R"))
+
+# Remove tidylog
+detach("package:tidylog", unload = TRUE)
+
+# Load basefile
+basefile <- read_rds(here::here("data", "basefiles", 
+                                glue("{pub_date}_base-file.rds")))
+
+# Basefile for most recent FY
+latest <- summarise_data(basefile)
+
+# Health Board data
+hb <- basefile %>% summarise_data(hb)
+max_hb <- hb %>% filter(qom == max(qom))
+min_hb <- hb %>% filter(qom == min(qom))
+
+# HSCP data
+hscp <- basefile %>% summarise_data(hscp)
+max_hscp <- hscp %>% filter(qom == max(qom))
+min_hscp <- hscp %>% filter(qom == min(qom))
+
+# Age Gender data
+allages <- basefile %>% summarise_data(sex)
+old     <- basefile %>% filter(age_grp == "85+" & !is.na(sex)) %>% summarise_data(sex)
+young   <- basefile %>% filter(age_grp == "0-54" & !is.na(sex)) %>% summarise_data(sex)
+
+max_old   <- old %>% filter(qom == max(qom)) %>% pull(sex) %>% tolower()
+min_old   <- old %>% filter(qom == min(qom)) %>% pull(sex) %>% tolower()
+max_young <- young %>% filter(qom == max(qom)) %>% pull(sex) %>% tolower()
+min_young <- young %>% filter(qom == min(qom)) %>% pull(sex) %>% tolower()
+
+# SIMD data
+simd <- basefile %>% summarise_data(simd)
+max_simd <- simd %>% filter(qom == max(qom)) %>% pull(simd)
+min_simd <- simd %>% filter(qom == min(qom)) %>% pull(simd)
+
+# Urban Rural data
+ur <- basefile %>% summarise_data(urban_rural) %>%
+  mutate(urban_rural = 
+           case_when(
+             str_detect(urban_rural, "Rural$") ~ paste(tolower(substring(urban_rural, 3)), "areas"),
+             TRUE ~ tolower(substring(urban_rural, 3))
+           ))
+max_ur <- ur %>% filter(qom == max(qom)) %>% pull(urban_rural)
+min_ur <- ur %>% filter(qom == min(qom)) %>% pull(urban_rural)
 
-```{r ListofTables, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE}
+```
+
+```{r links, echo = FALSE, message = FALSE, warning = FALSE, error = FALSE}
+
+data_tables <-
+  glue("http://www.isdscotland.org/Health-Topics/Health-and-Social-Community-Care/Publications/",
+       "{pub_date}/QOM_tables_{year(pub_date)}{format(pub_date, '%m')}.xls")
```

Comment:
I found it's more straightforward to use month() to extract the month number from pub_date. I've tested it and should work fine:

```
  glue("http://www.isdscotland.org/Health-Topics/Health-and-Social-Community-Care/Publications/",
       "{pub_date}/QOM_tables_{year(pub_date)}{month(pub_date)}.xls")
```

Reply:
Using format rather than month ensures you get a leading zero for single figure months; e.g. for the May pub we want 05, not 5.

Reply:
Ah okay that makes sense!